doc/classes/BaseMaterial3D.xml

```diff
@@ -445,22 +448,29 @@
 			Use [code]UV2[/code] with the detail texture.
 		</constant>
 		<constant name="TRANSPARENCY_DISABLED" value="0" enum="Transparency">
+			The material will not use transparency.
 		</constant>
 		<constant name="TRANSPARENCY_ALPHA" value="1" enum="Transparency">
+			The material will use the texture's alpha values for transparency.
 		</constant>
 		<constant name="TRANSPARENCY_ALPHA_SCISSOR" value="2" enum="Transparency">
 		</constant>
 		<constant name="TRANSPARENCY_ALPHA_DEPTH_PRE_PASS" value="3" enum="Transparency">
 		</constant>
 		<constant name="TRANSPARENCY_MAX" value="4" enum="Transparency">
+			Represents the size of the [enum Transparency] enum.
 		</constant>
 		<constant name="SHADING_MODE_UNSHADED" value="0" enum="ShadingMode">
+			The object will not receive shadows.
 		</constant>
 		<constant name="SHADING_MODE_PER_PIXEL" value="1" enum="ShadingMode">
+			The object will be shaded per pixel. Useful for realistic shading effect.
 		</constant>
 		<constant name="SHADING_MODE_PER_VERTEX" value="2" enum="ShadingMode">
+			The object will be shaded per vertex. Useful for toon shading effect.
```

Comment:
The object will be shaded per vertex. Useful when you want cheaper shaders and do not care about visual quality. 

I would drop the reference to toon shading as most toon shaders will still use per-pixel.